pkg/subroutines/extensions.go

```diff
@@ -157,34 +164,69 @@ func collectExtensions(ctx context.Context, cl client.Client, lookupNamespace st
 	return extensions, nil
 }
 
-func getParentAccount(ctx context.Context, cl client.Client, ns string) (*v1alpha1.Account, error) {
+func getParentAccount(ctx context.Context, cl client.Client, ns string) (*v1alpha1.Account, *string, error) {
+
+	if cluster, ok := kontext.ClusterFrom(ctx); ok && !cluster.Empty() {
+
+		wsCtx := kontext.WithCluster(ctx, logicalcluster.Name(""))
+		list := &tenancyv1alpha1.WorkspaceList{}
+
+		err := cl.List(wsCtx, list)
+		if err != nil {
+			return nil, nil, err
+		}
+
+		for _, ws := range list.Items {
+			if ws.Spec.Cluster != cluster.String() {
+				continue
+			}
+
+			clusterName := ws.Annotations[logicalcluster.AnnotationKey]
+
+			parentCtx := kontext.WithCluster(ctx, logicalcluster.Name(clusterName))
+
+			parentAccount := v1alpha1.Account{}
+			err = cl.Get(parentCtx, types.NamespacedName{
+				Name:      ws.Annotations[v1alpha1.NamespaceAccountOwnerLabel],
+				Namespace: ns,
```

Comment:
So this only works when all accounts are located in the same namespace within the workspaces right?

Reply:
Yes, an alternative would be to also include the namespace information in the workspace annotations. Could also be a valid option, as we need to add some information to the annotations anyway.

should I go ahead and add it?

Reply:
i'd suggest experiment with this now and add the namespace in a follow up PR